crates/bevy_ecs/src/world/mod.rs

```diff
@@ -591,64 +591,104 @@ impl World {
         }
     }
 
+    /// Inserts a new resource with default values.
+    ///
+    /// If the resource already exists, nothing happens.
+    ///
+    /// Uses the [`FromWorld`] trait to determine values,
+    /// which has a blanket impl for all `T: Default`.
+    #[inline]
+    pub fn init_resource<R: Resource + FromWorld>(&mut self) {
+        // PERF: We could avoid double hashing here, since the `from_resources` call is guaranteed
+        // not to modify the map. However, we would need to be borrowing resources both
+        // mutably and immutably, so we would need to be extremely certain this is correct
+        if !self.contains_resource::<R>() {
+            let resource = R::from_world(self);
+            self.insert_resource(resource);
+        }
+    }
+
     /// Inserts a new resource with the given `value`.
+    ///
     /// Resources are "unique" data of a given type.
+    /// Insert a resource of a type that already exists
```

Comment:
This reads a bit weird. Maybe "If a resource of type `R` already exists, its value will be overwritten."?